### ISSUE TITLE
Provide an AppImage for Linux

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#1060]
Carried over from https://github.com/brave/browser-laptop/issues/10250

## Description
> I would like to try out your browser but without letting it change repositories or libraries in my system.
> 
> Providing an [AppImage](http://appimage.org/) would have, among others, these advantages:
> - Applications packaged as an AppImage can run on many distributions (including Ubuntu, Fedora, openSUSE, CentOS, elementaryOS, Linux Mint, and others)
> - One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
> - No unpacking or installation necessary
> - No root needed
> - No system libraries changed
> - Works out of the box, no installation of runtimes needed
> - Optional desktop integration with `appimaged`
> - Optional binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
> - Can optionally GPG2-sign your AppImages (inside the file)
> - Works on Live ISOs
> - Can use the same AppImages when dual-booting multiple distributions
> 
> [Here is an overview](https://github.com/probonopd/AppImageKit/wiki/AppImages) of projects that are already distributing upstream-provided, official AppImages.
> 
> If you are interested, I am happy to help you make it happen. Just let me know.

## Latest status
@posix4e has set this up for `browser-laptop`; it was never officially released though

**Comment by probonopd** [comment]
Please let me know if I can assist you with this in any way. Does Brave build on Ubuntu 14.04 (trusty), the oldest currently still-supported LTS release of Ubuntu?

**Comment by bsclifton** [comment]
@probonopd it should build there without a problem 😄 Unlike `browser-laptop` before, the build process involves grabbing all of the Chromium code (which takes ~1 hour) and then compiling it all (which may take ~6 or more hours).  But you can check the binaries on our release page:
https://github.com/brave/brave-browser/releases

**Comment by azubieta** [comment]
Please :smile:

**Comment by probonopd** [comment]
Are the packages on https://github.com/brave/brave-browser/releases suitable for Ubuntu 14.04?

Then we could use https://github.com/AppImage/AppImages `pkg2appimage` to convert the deb to AppImage without having to recompile.

**Comment by bsclifton** [comment]
@probonopd yes- those should be fine. They're not signed though- if you want to get the signed packages, please check out the notes here:
https://brave-browser.readthedocs.io/en/latest/installing-brave.html#linux

**Comment by probonopd** [comment]
On that page it mentions Ubuntu 16.04+ though while I am looking to support all still-supported LTS releases of Ubuntu, which currently includes 14.04.

**Comment by Smooey** [comment]
Just curious, when using AppImages like browsers, can you still save your bookmarks, and passwords in it or not? I'm not familiar with how to create Void Linux packages but would love to be able to install Brave into Void Linux in near future. The only "privacy browser" I can find to install in Void Linux so far is an AppImage of "Iridium Browser". https://dl.bintray.com/probono/AppImages/

**Comment by posix4e** [comment]
18.04 is the current lts

**Comment by probonopd** [comment]
Yes, 14.04 is no longer supported. 16.04 (Xenial) is now the oldest still-supported LTS, and hence the one that should be considered the baseline target system.

**Comment by probonopd** [comment]
I am happy to help the Brave team make an AppImage. Just let me know.

**Comment by posix4e** [comment]
Amazing, I've been looking for help. Please email me at
alexnewmanbrave@wuli.nu and we can take it from there.

**Comment by posix4e** [comment]
Sorry for the delay. I talked to the brave people and we are good to hook up travis

- [ ] Hook up travis on a fork and verify it works
- [ ] Open a new issue to merge into the brave travis account
- [ ] Profit

**Comment by azubieta** [comment]
@posix4e let's know if you need some assistance.

**Comment by probonopd** [comment]
I have started working on it but keep running into issues building Brave (even before beginning to AppImage-ify it). Any hints appreciated.

* Travis build script: https://github.com/probonopd/brave-browser/blob/patch-1/.travis.yml
* Travis build log: https://travis-ci.com/probonopd/brave-browser

**Comment by posix4e** [comment]
Travis support helping at all?

**Comment by mihaiplesa** [comment]
@hcsubser added this recently https://github.com/hcsubser/Brave-AppImage and there's one release there

**Comment by hcsubser** [comment]
> @hcsubser added this recently https://github.com/hcsubser/Brave-AppImage and there's one release there

There is now a better unofficial appimage of brave at: https://github.com/srevinsaju/Brave-AppImage which is also fully automatic but covers all versions and not just bleeding edge versions of brave like mine

**Comment by probonopd** [comment]
Wuold still be good to have an official one since running (no offense!) binaries from "random third parties" is generally discouraged for security reasons; especially for a web browser.

**Comment by posix4e** [comment]
Perhaps the better upstream would be willing to submit it as a patch and be
a part of the official brave process?

**Comment by mihaiplesa** [comment]
We're happy to take this in at Brave similar to how it was done for snaps.

**Comment by probonopd** [comment]
@azubieta do you think you could have a look at this?

**Comment by azubieta** [comment]
Added to my TODO

**Comment by 0-5-0** [comment]
Any luck with this? It would be great to have the Brave browser in appimage format and from a trusted source.

**Comment by wknapik** [comment]
@0-5-0 there's an official Snap and an unofficial Flatpak (see [brave.com/linux](https://brave.com/linux/)).

I don't think it makes sense for Brave to maintain an AppImage. It's the least popular of the 3 and AppImage itself hasn't had a new release in 2 years. 

IMO this issue should be closed as WONTFIX.

**Comment by Samueru-sama** [comment]
Hey since brave already releases a zipped portable build, you can just turn that into an appimage, which is already done [here](https://github.com/srevinsaju/Brave-AppImage) for example.

The appimage is just a more convenient way to distribute portable builds. 

The only detail is that if you go with this approach, please use this appimagetool instead as it uses the static appimage runtime (works on distros that only have fuse3 instead of fuse2): https://github.com/probonopd/go-appimage

**Comment by Xmetalfanx** [comment]
I would love to see an official Appimage.   Any chance (reading through this issue fast but i am not sure how many third party releases for AppImages there may be out there) that if they do good work, the third party people doing those releases could help out or even become (even if just for making the appimage or automating it based on new Brave releases) part of the team and those AppImages could be listed here as now "official" releases?

**Comment by ivan-hc** [comment]
### Dear Brave community... dear Brave team,

Between the lines, I can see why an AppImage is the least choice by the Brave team.

----------------------------------------
### Problem 1: the problem of every browser
A browser MUST be up to date with security fixes, and this can only be done through centralization... and Flatpak and Snap officially guarantee this.

Currently, AppImages can be updated via external tools, two GUI apps above all:
- the most used is available as AppImage, DEB and RPM, and requires modifying some system files in order to make its daemon work to detect the presence of AppImage packages in the system (compromising its use from the command line, for example... problem never solved)
- the second is released as a Flatpak only, so those who use Flatpak are forced to use the Brave Flatpak and are not encouraged to use the AppImage

That said, AppImages not have a centralized method to be distributed and updated...

<details>
  <summary>...or at least not officially (SPOILER).</summary>

I myself, who for 4 years have been building and maintaining a package manager for AppImages that I called "[AM](https://github.com/ivan-hc/AM)" admit (as @probonopd  always points out) that providing a centralized method to distribute AppImages makes them lose the very sense of the reason they were conceived (i.e. put them where you want and use them), by converting the AppImages usage to the use of any Snap, Flatpak, RPM or DEB package: in one place!

Well, why do I maintain a package manager for AppImages? Because the AppImages have many other qualities, and two of these above all:
1. they are compressed (if Brave takes up 300 MB, an AppImage takes up half of it)
2. they are isolable (creating a .home or .config directory near them isolates the dotfiles)

finally, making them centralized for my part makes them even better manageable (backup, restore, sandboxing, downgrade...)... **but this is my use case**, if a gun dealer sells me a gun, it is obvious that if I shoot I will be thrown in jail, this does not mean that the person who sold it to me should tell me "please, do not shoot anyone" (it is the same concept of the Terms of Service introduced in Firefox). In my case the gun dealer is the inventor of AppImage.

</details>

...**and this is the main problem.**

----------------------------------------
### Problem 2: packaging
Packaging on the other hand is absolutely not a problem! **Anyone can do it!**

You have already mentioned the fantastic work done [here](https://github.com/srevinsaju/Brave-AppImage) for the x86_64 builds

Well, I also wrote a script to build the builds for x86_64 and aarch64 starting from the official .zip packages present in releases, at https://github.com/ivan-hc/Brave-appimage. You can do it too:

https://github.com/ivan-hc/Brave-appimage/blob/main/brave-builder.sh

...the only third-party component that has been added is [the .desktop file](https://github.com/ivan-hc/Brave-appimage/blob/main/brave.desktop), but only because I did not find it on this repository.

Finally, both for my AppImages and for [srevinsaju](https://github.com/srevinsaju)'s, the dependency on libfuse2 is no longer there (the whole AppImage ecosystem has improved in this sense), and they are all updatable using `zsync` and `appimageupdatetool`.

----------------------------------------
### Final conclusions
NOW, I address the Brave team directly... how much would it cost you to run a single script that repackages the builds from ZIP to AppImage? How much effort and manpower does it take?

You have [a huge list of artifacts for each release](https://github.com/brave/brave-browser/releases/latest)... is it possible that an AppImage costs you so much?

And mind you, **I'm not a Brave user**, I don't care about using one browser or the other... **I speak for AppImage users**, who are many, and **whose packaging is really TOO EASY TO MAKE!**

### If the basic problem is not centralized distribution, then I don't understand you.

Greetings.
